cmd/podman/images/save.go

```diff
@@ -91,11 +96,33 @@ func save(cmd *cobra.Command, args []string) error {
 		return errors.Errorf("--compress can only be set when --format is either 'oci-dir' or 'docker-dir'")
 	}
 	if len(saveOpts.Output) == 0 {
-		fi := os.Stdout
-		if terminal.IsTerminal(int(fi.Fd())) {
-			return errors.Errorf("refusing to save to terminal. Use -o flag or redirect")
+		pipeFile := "/tmp/saveio"
+		saveOpts.Output = pipeFile
+		err := os.Remove(pipeFile)
+		if err != nil && !os.IsNotExist(err) {
+			return errors.Wrapf(err, "error removing named pipe")
 		}
-		saveOpts.Output = "/dev/stdout"
+		err = syscall.Mkfifo(pipeFile, 0666)
+		if err != nil {
+			return errors.Wrapf(err, "error creating named pipe")
+		}
+		defer os.Remove(pipeFile)
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			fpipe, _ := os.OpenFile(pipeFile, os.O_RDWR, os.ModeNamedPipe)
+			defer fpipe.Close()
+			for {
```

Comment:
How does this loop ever terminate? Not on EOF, not on removing `pipeFile`. So, only on process exit, I think — but the main goroutine does not know when this one is done.